pkg/metrics/metrics.go

```diff
@@ -23,20 +23,62 @@ import (
 	"sync"
 	"time"
 
+	crdv1 "github.com/kubernetes-csi/external-snapshotter/client/v3/apis/volumesnapshot/v1beta1"
 	"github.com/prometheus/client_golang/prometheus/promhttp"
 	"k8s.io/apimachinery/pkg/types"
 	k8smetrics "k8s.io/component-base/metrics"
 	klog "k8s.io/klog/v2"
 )
 
 const (
-	opStatusUnknown      = "Unknown"
-	labelDriverName      = "driver_name"
-	labelOperationName   = "operation_name"
-	labelOperationStatus = "operation_status"
-	subSystem            = "snapshot_controller"
-	metricName           = "operation_total_seconds"
-	metricHelpMsg        = "Total number of seconds spent by the controller on an operation from end to end"
+	labelDriverName                    = "driver_name"
+	labelOperationName                 = "operation_name"
+	labelOperationStatus               = "operation_status"
+	labelSnapshotType                  = "snapshot_type"
+	subSystem                          = "snapshot_controller"
+	latencyReconciliationMetricName    = "reconciliation_total_seconds"
+	latencyReconciliationMetricHelpMsg = "Number of seconds spent by the controller on intermediate reconciliations"
+	latencyOperationMetricName         = "operation_total_seconds"
+	latencyOperationMetricHelpMsg      = "Total number of seconds spent by the controller on an operation from end to end"
+
+	// CreateSnapshotOperationName is the operation that tracks how long the controller takes to create a snapshot.
+	// Specifically, the operation metric is emitted based on the following timestamps:
+	// - Start_time: controller notices the first time that there is a new VolumeSnapshot CR to dynamically provision a snapshot
```

Comment:
The start time description is relevant to the e2e metric, but not the reconcilation one. Can we clarify these comments